test/unit-test/test_lib/test_vis.py

```diff
@@ -29,7 +32,20 @@ def test_calc_plot_linear_fit(self, x_in, y_in, expected_result):
         os.remove(current_dir + '/linear_plot_test.png')
         
         assert np.allclose(output, expected_result)
-    
+
```

Comment:
Mocks in Python has a unique problem vis-a-vis mocks in Java. In Java, any type signature violations are caught by Compiler; With Python, there is no such protection.

The type mismatch between mock and the actual type implementation are detected through [autospeccing](https://docs.python.org/3/library/unittest.mock.html#auto-speccing). Please update the unit test with autospeccing protection so that any changes to the original function leads to failure of the unit test (a much needed protection).

Reply:
Done

Reply:
As far as (d for d in data) is concerned, I used it in conformation with the [vis.py](https://github.com/bhuvangupta008/IRCLogParser/blob/bhuvan-test/lib/vis.py#L26) file as used in the project. 
Alternative would have been, accessing the tuple members via index like 
```
x_data = data[0]
y_data = data[1]
```